.github/workflows/tests.yml

```diff
@@ -14,18 +14,20 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        resolver: [lts-16, lts-15, lts-14]
+        resolver: [lts-16, lts-15, lts-14, lts-12]
+        yaml: ['stack.yaml', 'stack-master.yaml']
 
     steps:
       - name: Clone project
         uses: actions/checkout@v2
 
       - name: Cache dependencies
-        uses: actions/cache@v1
+        uses: actions/cache@v2
         with:
           path: ~/.stack
-          key: ${{ runner.os }}-${{ matrix.resolver }}-${{ hashFiles('stack.yaml') }}
+          key: ${{ runner.os }}-${{ matrix.resolver }}-${{ hashFiles(matrix.yaml) }}-${{ hashFiles('xmonad-contrib.cabal') }}
           restore-keys: |
+            ${{ runner.os }}-${{ matrix.resolver }}-${{ hashFiles(matrix.yaml) }}
```

Comment:
It's probably not important but can we be consistent here and add a trailing `-` to the first `restore-keys:` entry just like below? :-)

Reply:
Done.